rust/src/oid4vp/dc_api/prepare_response.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::{Context, Result};
+use base64::prelude::*;
+use isomdl::{
+    cbor,
+    definitions::{helpers::ByteStr, DeviceResponse},
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value as Json;
+use sha2::{Digest, Sha256};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Handover(String, ByteStr);
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HandoverInfo(String, String, String);
+
+impl Handover {
+    pub fn new(origin: String, client_id: String, nonce: String) -> Result<Self> {
+        let handover_info = HandoverInfo(origin, client_id, nonce);
+        let handover_info_bytes = cbor::to_vec(&handover_info)?;
+        let handover_info_hash = ByteStr::from(Sha256::digest(handover_info_bytes).to_vec());
+        Ok(Handover(
+            "OpenID4VPDCAPIHandover".to_string(),
+            handover_info_hash,
+        ))
+    }
+}
```

Comment:
Does it make sense to combine these with the `Handover::OID4VP` type in `isomdl` crate?

Reply:
I don't think so, probably we should just remove the `Handover::OID4VP` variant from `isomdl`. We possibly need to make some decisions about what the responsibilities of the different libraries are, but IMO `isomdl` is currently:
* Data structures according to 18013-5.
* 18013-5 presentation transaction logic.

The OpenID4VP stuff is constantly changing, so for now I would prefer to keep it separate.